src/test/java/uk/gov/pay/webhooks/deliveryqueue/managed/WebhookMessagePollingServiceTest.java

```diff
@@ -0,0 +1,138 @@
+       verifyNoInteractions(webhookMessageSenderMock);
+    }
+
+    @Test
+    public void shouldEmitSingleValidQueueItem() throws IOException, InvalidKeyException, InterruptedException {
+        setupOrderedDeliveryQueueWith(List.of("first-external-id"));
+        webhookMessagePollingService.pollWebhookMessageQueue();
+        verify(webhookMessageSenderMock).sendWebhookMessage(argThat((webhookMessage -> webhookMessage.getExternalId().equals("first-external-id"))));
+    }
+
+    @Test
+    public void shouldEmitMultipleValidQueueItemsInOnePoll() throws IOException, InvalidKeyException, InterruptedException {
+        setupOrderedDeliveryQueueWith(List.of("first-external-id", "second-external-id"));
+        webhookMessagePollingService.pollWebhookMessageQueue();
+
+        var captor = ArgumentCaptor.forClass(WebhookMessageEntity.class);
+        verify(webhookMessageSenderMock, times(2)).sendWebhookMessage(captor.capture());
+        assertEquals("first-external-id", captor.getAllValues().get(0).getExternalId());
+        assertEquals("second-external-id", captor.getAllValues().get(1).getExternalId());
+    }
+
+    @Test
+    public void shouldAppropriatelyHandleFailedEmitAndNotTryAgainImmediately() throws IOException, InvalidKeyException, InterruptedException {
+        when(response.statusCode()).thenReturn(500);
+
+        setupOrderedDeliveryQueueWith(List.of("first-external-id"));
+        webhookMessagePollingService.pollWebhookMessageQueue();
+        verify(webhookMessageSenderMock).sendWebhookMessage(argThat((webhookMessage -> webhookMessage.getExternalId().equals("first-external-id"))));
+
+        webhookMessagePollingService.pollWebhookMessageQueue();
+        verifyNoMoreInteractions(webhookMessageSenderMock);
+    }
+
+    private void setupOrderedDeliveryQueueWith(List<String> messageIds) {
+        WebhookEntity webhookEntity = new WebhookEntity();
+
+        database.inTransaction(() -> {
+            webhookEntity.setCallbackUrl("http://a-callback-url.com");
+            webhookEntity.setServiceId("a-service-id");
+            webhookEntity.setLive(false);
+
+            webhookDao.create(webhookEntity);
+        });
+
+        for (int i = 0; i < messageIds.size(); i++) {
+            int finalI = i;
+
+            var message = database.inTransaction(() -> {
+                WebhookMessageEntity webhookMessageEntity = new WebhookMessageEntity();
+                var id = messageIds.get(finalI);
+                webhookMessageEntity.setExternalId(id);
+                webhookMessageEntity.setWebhookEntity(webhookEntity);
+                webhookMessageEntity.setCreatedDate(instantSource.instant());
+
+                return webhookMessageDao.create(webhookMessageEntity);
+            });
+            database.inTransaction(() -> {
+                webhookDeliveryQueueDao.enqueueFrom(message, WebhookDeliveryQueueEntity.DeliveryStatus.PENDING, instantSource.instant().minusMillis((messageIds.size() - finalI) * 10L));
```

Comment:
I wonder if it might be clearer to have a mock `InputSource` and just stub it to return the desired value each time.